lib/jumpy-view.coffee

```diff
@@ -40,58 +42,61 @@ class JumpyView extends View
-
-        isMatchOfCurrentLabels = (character, labelPosition) =>
-            found = false
-            @disposables.add atom.workspace.observeTextEditors (editor) =>
-                editorView = atom.views.getView(editor)
-                return if $(editorView).is ':not(:visible)'
-
-                for decoration in @decorations
-                    element = decoration.getProperties().item
-                    if element.textContent[labelPosition] == character
-                        found = true
-                        return false
-            return found
-
-        # Assert: labelPosition will start at 0!
-        labelPosition = (if not @firstChar then 0 else 1)
-        if !isMatchOfCurrentLabels character, labelPosition
-            @statusBarJumpy?.classList.add 'no-match'
-            @statusBarJumpyStatus?.innerHTML = 'No match!'
-            return
-
-        if not @firstChar
-            @firstChar = character
-            @statusBarJumpyStatus?.innerHTML = @firstChar
-            # TODO: Refactor this so not 2 calls to observeTextEditors
-            @disposables.add atom.workspace.observeTextEditors (editor) =>
-                editorView = atom.views.getView(editor)
-                return if $(editorView).is ':not(:visible)'
-
+        mobx.useStrict(true)
+        @mobJumpy = mobx.observable(
+            # global
+            statusBarJumpy: @statusBarJumpy
+
+            # Labels -----------------
+            allPositions: mobx.asMap({})
+            setPositions: ->
+                mobx.action (key, value) =>
+                    @allPositions.set key, value
+            relevantPositions: -> # not necessarily reflected whith a 'no match'
+                mobx.computed =>
+                    mobx.asMap(@allPositions.entries().filter ([key]) =>
+                        key.startsWith @currentKeys)
+            clearAllPositions: mobx.action ->
+                @allPositions.clear()
+
+            # Input -----------------
+            currentKeys: ''
+            statusMessage: 'Jump Mode!'
+            acceptKey: ->
+                mobx.action (character) =>
+                    @currentKeys += character
+
+                    @statusBarJumpy?.classList.remove 'no-match'
+                    @setStatusMessage @currentKeys
+                    if !@isMatch.get()
+                        @statusBarJumpy?.classList.add 'no-match'
+                        @setStatusMessage 'No match!'
+            isMatch: ->
+                mobx.computed =>
+                    @allPositions.entries().find ([key]) =>
+                        key.startsWith @currentKeys
+            removeKey: ->
+                mobx.action ->
+                    @currentKeys = @currentKeys[...-1]
+            setStatusMessage: ->
+                mobx.action (message) ->
+                    @statusMessage = message
+            reset: mobx.action =>
+                @statusBarJumpy?.classList.remove 'no-match'
```

Comment:
I can't just set no-match on when length == 0, because I can get there from different ways. I thought about simplifying the problem with a timed.. "No Match!" goes away after a while. But I think I can work with the above concept a bit more